MM-17912 - slash command to exercise FileInfo search API

ashishbhate opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does.
-->
- Demonstrate usage of the `GetWithOptions` method to get FileInfos that was added in https://github.com/mattermost/mattermost-server/pull/13647/ 

#### Ticket Link
<!--
If this pull request addresses a Help Wanted ticket, please link the relevant GitHub issue, e.g.

  Fixes https://github.com/mattermost/mattermost-server/issues/XXXXX

Otherwise, link the JIRA ticket.
-->
- Part of https://github.com/mattermost/mattermost-server/issues/12699 and
https://mattermost.atlassian.net/browse/MM-17912

ashishbhate commented:
CI will fail because https://github.com/mattermost/mattermost-server/pull/13647/  hasn't been merged yet. 

On my local machine I have updated the `go.mod` file to point to my fork of the mattermost-server that has the required changes. You can do this by running
`go mod edit -replace=github.com/mattermost/mattermost-server/v5=github.com/ashishbhate/mattermost-server/v5@25037fd7`

hanzei commented:
@ashishbhate Yes, please update `go.mod` to make the build pass. :+1:

ashishbhate commented:
@hanzei I've updated the `go.mod` file.

ali-farooq0 commented on server/command_hooks.go:
```diff
@@ -509,3 +521,60 @@ func (p *Plugin) crash() {
+		attachments = append(attachments,
+			&model.SlackAttachment{
+				Title:     f.Name,
+				TitleLink: permaLink + f.PostId,
```
this is really cool! It'll be great to include this file's public url as well. As I imagine most people using this api would want to display it's public url, and looking at `FileInfo` struct it's not very clear how would one go about doing that, so it'll be great to include an example here.

ashishbhate replied:
Do you mean in addition to the link to the post Id or instead of the link to the post id?

ali-farooq0 replied:
in addition works, looks like you already made the change, excellent! 👌

ashishbhate commented:
I've updated the PR based on your feedback:

1. https://github.com/mattermost/mattermost-server/pull/13647 has been merged so I've updated the `go.mod` file to use upstream mattermost-server instead of my fork. There were some unrelated dependency updates as well. 

2. I've updated the slash command to show the file's public link. The server config will need to be set to allow public links: FileSettings>EnablePublicLink
I've made this a required configuration in this plugin's manifest

3. I've updated the README to include the added slash command

mattermod commented:
This issue has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

/cc @jasonblais @hanzei

jasonblais commented:
@ali-farooq0 Reminder to help with your review when you're back :)

ali-farooq0 commented:
@prapti this is ready for testing

ali-farooq0 commented:
/update-branch
